source/include/drivers/cs_RTC.h

```diff
@@ -18,61 +21,127 @@
 /** Maximum value of the RTC counter. */
 #define MAX_RTC_COUNTER_VAL 0x00FFFFFF
 
-/** Wrapper class for RTC functions.
+/**
+ * Wrapper class for RTC functions.
  *
- * if NRF51_USE_SOFTDEVICE==1 uses the RTC0 clock
- * managed by the softdevice
+ * if NRF51_USE_SOFTDEVICE==1 uses the RTC0 clock managed by the softdevice,
  * otherwise uses the RTC1 clock used by the app timer
- *
  * With the current clock frequency and max counter val, the RTC overflows every 512s.
  *
- * TODO: Implementation should be in a .cpp file
- * TODO: Get rid of all these static functions.
- * TODO: We have hardware floating points now. Get rid of ROUNDED_DIV macro, etc.
  */
 class RTC {
-
 private:
-	RTC(){};
-	RTC(RTC const&);             //! singleton, deny implementation
-	void operator=(RTC const&);  //! singleton, deny implementation
+	RTC()           = default;
+	RTC(RTC const&) = delete;
+	void operator=(RTC const&) = delete;
 
 public:
-	//! return number of ticks
-	inline static uint32_t getCount() { return NRF_RTC0->COUNTER; }
 
-	//! return difference between two tick counter values
-	inline static uint32_t difference(uint32_t ticksTo, uint32_t ticksFrom) {
-		return ((ticksTo - ticksFrom) & MAX_RTC_COUNTER_VAL);
-	}
+	/**
+	 * return number of ticks
+	 */
+	INLINE static uint32_t getCount();
```

Comment:
What's with the uppercase inline? :P
I'm not sure why these have to be inline at all actually.

Reply:
done, it's in a .cpp file now so the INLINEs have been removed